### DISCUSSION TITLE
archi di peso diverso

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/progettazione-di-algoritmi#45] [Esercizi A-L]
Dato un grafo $G$ e $p : E(G) \to \mathbb{R}^+$, con $i \neq j \implies p(i) \neq p(j)$ _(tutti gli archi hanno peso diverso)_, dimostrare che $\exists !$ un minimum spanning tree.

**Comment by cornflexxx** [comment]
Consideriamo l' array ordinato in base al peso di tutti gli archi di $G$, $S(E(G)) := [e_1,e_2,...,e_n | e_1\lt e_2\lt ...\lt e_n]$. Quindi prendendo in ordine ciascuno degli archi costruiamo con l'algoritmo di $Kruskal$ in modo univoco un albero. Supponiamo però di prendere alcuni degli archi che vengono scartati dall'algoritmo perche contribuiscono a creare un ciclo, se proseguissimo con questa strategia il peso dell'albero che stiamo costruendo sarebbe maggiore dell' $MST$ costruito procedendo normalmente: infatti l'arco del ciclo che con $Kruskal$ tralasciamo sarà l'arco più pesante del ciclo stesso. Ne segue quindi che l'albero di copertura minima è unico se i pesi degli archi sono tutti diversi.
